src/fsharp/FSharp.Compiler.Private.Scripting/FSharpScript.fs

```diff
@@ -7,30 +7,9 @@ open System.Threading
 open FSharp.Compiler
 open FSharp.Compiler.Interactive.Shell
 
-type FSharpScript(?captureInput: bool, ?captureOutput: bool, ?additionalArgs: string[]) as this =
-    let outputProduced = Event<string>()
-    let errorProduced = Event<string>()
+type FSharpScript(?additionalArgs: string[]) as this =
```

Comment:
please don't use as **this** typically using **this** in a type constructor is considered a bit iffy.

I think you can replace
````
this.GetType().Assembly.Location;
````
with:
````
typeof<FSharpScript>.Assembly.Location
````